app/web/src/newhotness/explore_grid/ExploreGridRow.vue

```diff
@@ -276,10 +316,56 @@ const props = defineProps<{
 const exploreContext = inject<ExploreContext>("EXPLORE_CONTEXT");
 assertIsDefined<ExploreContext>(exploreContext);
 
-const renderLastPath = (path: string) => {
-  const last = path.split("/").at(-1) ?? "unknown";
-  return last.charAt(0).toLocaleUpperCase() + last.slice(1);
-};
+const ctx = useContext();
+
+const key = useMakeKey();
+const args = useMakeArgs();
+
+const enableDefaultSubSourceValuesQuery = computed(
+  () => ctx.queriesEnabled && props.row.type === "defaultSubHeader",
+);
+const defaultSubSourceValuesQuery = useQuery<
+  { prop: string; value: string } | undefined
+>({
+  enabled: enableDefaultSubSourceValuesQuery,
+  queryKey: key(
+    EntityKind.AttributeTree,
+    props.row.type === "defaultSubHeader" ? props.row.componentId : "",
+    "DefaultSubHeader",
+  ),
+  queryFn: async () => {
+    if (props.row.type !== "defaultSubHeader") {
+      return undefined;
+    }
+    const attributeTree = await bifrost<AttributeTree>(
+      args(EntityKind.AttributeTree, props.row.componentId),
+    );
+    const value =
```

Comment:
im wary of modifying data within the `queryFn`... its safe to do when the `queryFn` is idempotent for every state / input of the `queryKey`...

What tips me off that it _may_ not be true is the inclusion of `props.row.path` in the `queryFn`, but not in the `queryKey`

Reply:
Will never change... based on what? The _actual_ default sub?